sentry_sdk/integrations/redis/__init__.py

```diff
@@ -155,15 +158,20 @@ def _set_cluster_db_data(span, redis_cluster_instance):
 
 
 def _set_async_cluster_db_data(span, async_redis_cluster_instance):
-    # type: (Span, AsyncRedisCluster) -> None
+    # type: (Span, AsyncRedisCluster[Any]) -> None
     default_node = async_redis_cluster_instance.get_default_node()
     if default_node is not None and default_node.connection_kwargs is not None:
         _set_db_data_on_span(span, default_node.connection_kwargs)
 
 
 def _set_async_cluster_pipeline_db_data(span, async_redis_cluster_pipeline_instance):
-    # type: (Span, AsyncClusterPipeline) -> None
-    _set_async_cluster_db_data(span, async_redis_cluster_pipeline_instance._client)
+    # type: (Span, AsyncClusterPipeline[Any]) -> None
+    _set_async_cluster_db_data(
+        span,
+        # the AsyncClusterPipeline has always had a `_client` attr but it is private so potentially problematic and mypy
+        # does not recognize it - see https://github.com/redis/redis-py/blame/v5.0.0/redis/asyncio/cluster.py#L1386
+        async_redis_cluster_pipeline_instance._client,  # type: ignore[attr-defined]
```

Comment:
Is there a way we could try to avoid using `_client` here, and instead access what we need through a public API? I would strongly prefer to avoid using an external library's private API, since the private API could change at any time and cause problems in the future.

Reply:
Okay, in that case, we can use `capture_internal_exceptions` here. I also checked that one of your unit tests is verifying this function, so we will get a test failure if `_client` ever gets removed